test/seqan3-test.cmake

```diff
@@ -162,10 +163,7 @@ macro (seqan3_require_test)
     add_library (gtest STATIC IMPORTED)
     add_dependencies (gtest gtest_main)
     set_target_properties (gtest PROPERTIES IMPORTED_LOCATION "${gtest_library}")
-
-    list (APPEND SEQAN3_TEST_LIBRARIES "gtest_main")
-    list (APPEND SEQAN3_TEST_LIBRARIES "gtest")
-    list (APPEND SEQAN3_TEST_LIBRARIES "pthread")
+    set_property (TARGET gtest APPEND PROPERTY INTERFACE_LINK_LIBRARIES "pthread")
 endmacro ()
 
 macro (add_subdirectories)
```

Comment:
I am still a little confused about the entities in this file and naming. Is "benchmark" actually a "dependency handler" for "google_benchmark"? Could we rename it gbenchmark, like for "gtest"?

Could we also rename  our new "interface names" to be consistent with the filesystem hierarchy, i.e.
* `seqan3` instead of seqan::seqan3
* `seqan3_test` (stays the same)
* `seqan3_test_unit` instead of seqan3_unit_test
* `seqan3_test_performance` instead of seqan3_performance_test

Reply:
And should seqan3_test_unit and seqan3_test_performance not be moved to their respective folders' CMakeLists.txt?

Reply:
> I am still a little confused about the entities in this file and naming. Is "benchmark" actually a "dependency handler" for "google_benchmark"? Could we rename it gbenchmark, like for "gtest"?
Yep good idea!

> Could we also rename  our new "interface names" to be consistent with the filesystem hierarchy, i.e.
> * `seqan3` instead of seqan::seqan3
> * `seqan3_test` (stays the same)
> * `seqan3_test_unit` instead of seqan3_unit_test
> * `seqan3_test_performance` instead of seqan3_performance_test

Good idea, I would stay with `seqan3::seqan3`, since every recent cmake-module which exposes a 'interface library' does it through a namespaced one. For example Mpi exposes `mpi::mpi`.